cudaextender/include/claraparabricks/genomeworks/cudaextender/extender.hpp

```diff
@@ -0,0 +1,151 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+#include <claraparabricks/genomeworks/cudaextender/cudaextender.hpp>
+#include <claraparabricks/genomeworks/types.hpp>
+#include <claraparabricks/genomeworks/utils/allocator.hpp>
+#include <cuda_runtime_api.h>
+#include <vector>
+#include <memory>
+
+namespace claraparabricks
+{
+namespace genomeworks
+{
+namespace cudaextender
+{
+
+/// \addtogroup cudaextender
+/// \{
+
+/// Seed positions in target and query reads
+struct SeedPair
+{
+    /// position of first sketch element in query_read_id_
+    position_in_read_t query_position_in_read;
+    /// position of second sketch element in target_read_id_
+    position_in_read_t target_position_in_read;
+};
+
+/// Segment pairs in target & query reads with associated length & score
+struct ScoredSegmentPair
+{
+    /// seed for the segment
+    SeedPair seed_pair;
+    /// length of the segment
+    int32_t length;
+    /// score of the segment
+    int32_t score;
+};
+
+/// Equality operator for ScoredSegmentPair
+__host__ __device__ inline bool operator==(const ScoredSegmentPair& x, const ScoredSegmentPair& y)
+{
+    return ((x.seed_pair.target_position_in_read == y.seed_pair.target_position_in_read) && (x.seed_pair.query_position_in_read == y.seed_pair.query_position_in_read) && (x.length == y.length) && (x.score == y.score));
```

Comment:
minor - is there any way to make this statement multi line while keeping the linter happy?

Reply:
I wrestled with it for a bit. Couldn't coax it into a multi-line statement.

Reply:
No problem, thanks for trying :).